subprojects/diagnostics/src/main/java/org/gradle/api/reporting/dependencies/HtmlDependencyReportTask.java

```diff
@@ -66,7 +75,8 @@
  */
 @UntrackedTask(because = "We can't describe the dependency tree of all projects as input")
 public abstract class HtmlDependencyReportTask extends ConventionTask implements Reporting<DependencyReportContainer> {
-    private Set<Project> projects;
+    private Transient<SetProperty<Project>> projects = Transient.of(getObjectFactory().setProperty(Project.class).convention(Providers.of(getProject()).map(it -> Collections.singleton(it))));
```

Comment:
❓ What's the benefit of using a `SetProperty` here?

Reply:
"Represents a property whose type is a Set of elements of type T."

What is your concern? I remember you mentioning something about additional overhead but do not recall the details so went ahead with the nicer API.

Reply:
Or are you questioning using a `Property` altogether?

Reply:
If it is the latter, I just went followed the example of another task. I guess I could have just:

```
private Transient.Var<Set<Project>> projects = Transient.varOf(uncheckedCast(singleton(getProject())));
```

?

Reply:
Exactly.